### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by anatolii-zarytskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/GetArea.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface GetArea {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/GetArea.java:3]: please don't use verbs in class/interface names
```suggestion
public interface AreaCalculator {
```

File: src/main/java/core/basesyntax/GetArea.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    int areaCalculate();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/GetArea.java:4]:
```suggestion
    double calculateArea();
```

File: src/main/java/core/basesyntax/Draw.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Draw {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Draw.java:3]:
```suggestion
public interface Drawable {
```

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int sideA;
+    private int sideB;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:5]:
```suggestion
    private int firstBase;
    private int secondBase;
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private int sideA;
+    private int sideB;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java:5]:
```suggestion
    private int width;
    private int length;
```

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private int sideA;
+    private int sideB;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/RightTriangle.java:5]:
```suggestion
    private int firstLeg;
    private int secondLeg;
```

File: src/main/java/core/basesyntax/Main.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[new Random().nextInt(5)];
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java:8]: 5 is a magic number

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static Random random = new Random();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: why static?

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    private static final String[] COLORS = new String[] {
+            "red", "blue", "white", "brown", "green", "pink", "yellow", "orange"
+    };
+
+    public static String colorFigures() {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ColorSupplier.java:11]: same - please don't create static methods - they are in general bad practice
also, let's use verbs for method names
```suggestion
    public String getRandomColor() {
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_NUMBER = 5;
+    private static final int MAX_NUMBER = 10;
+    private Random random = new Random();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FigureSupplier.java:8]: create a ColorSupplier field here as well and call its methods through this field

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Main.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Main {
+
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java:6]: redundant empty line

File: src/main/java/core/basesyntax/Main.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        int maxSize = new Random().nextInt(100);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java:9]: 100 is a magic number - create a constant please

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by Resci — APPROVED** [review]

**Pull request closed without merging**